### PR TITLE
Workspaces application - deploy action

### PR DESCRIPTION
**Pull request by sfc-gh-gbloom**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description

Implementing `ApplicationEntity.action_deploy()` by extracting the logic from `NativeAppRunProcessor`.

### Manual Test
Create flow:
```
$ snow ws deploy --entity-id=app

Checking if stage pdfv2_pkg_gbloom.app_src.stage exists, or creating a new one if none exists.
Performing a diff between the Snowflake stage and your local deploy_root ('~/pdfv2/output/deploy') directory.
Local changes to be deployed:
  added:    app/README.md -> README.md
  added:    app/manifest.yml -> manifest.yml
  added:    app/setup_script.sql -> setup_script.sql

Deleted paths to be removed from your stage:
  deleted:  app/README.md
  deleted:  app/manifest.yml
  deleted:  app/setup_script.sql
Updating the Snowflake stage from your local ~/pdfv2/deploy directory.
Validating Snowflake Native App setup script.
Creating new application object pdfv2_gbloom in account.
Application 'PDFV2_GBLOOM' created successfully.

Deployed successfully.
```
Upgrade flow:
```
$ snow ws deploy --entity-id=app

Checking if stage pdfv2_pkg_gbloom.app_src.stage exists, or creating a new one if none exists.
Performing a diff between the Snowflake stage and your local deploy_root ('~/pdfv2/output/deploy') directory.
Your stage is up-to-date with your local deploy root.
Validating Snowflake Native App setup script.
Upgrading existing application object pdfv2_gbloom.
Application successfully upgraded.

Deployed successfully.
```

### CODE REVIEW

File: src/snowflake/cli/_plugins/workspace/action_context.py [hunk 1]
```diff
@@ -15,3 +15,4 @@ class ActionContext:
     project_root: Path
     default_role: str
     default_warehouse: Optional[str]
+    get_entity: Callable
```
Comment (sfc-gh-fcampbell) [inline src/snowflake/cli/_plugins/workspace/action_context.py:18]:
```suggestion
    get_entity: Callable[[str], EntityBase | None]
```

File: tests_integration/nativeapp/test_init_run.py [hunk 2]
```diff
@@ -38,7 +46,7 @@ def test_nativeapp_init_run_without_modifications(
 ):
     project_name = "myapp"
     with nativeapp_project_directory(test_project):
-        result = runner.invoke_with_connection_json(["app", "run"])
+        result = runner.invoke_with_connection_json(command.split())
```
Comment (sfc-gh-fcampbell) [inline tests_integration/nativeapp/test_init_run.py:49]: you imported `shlex.split` but it's not used here?

Comment (sfc-gh-gbloom) [reply tests_integration/nativeapp/test_init_run.py]: Fixed

File: src/snowflake/cli/_plugins/workspace/action_context.py (hunk 1 above)
Comment (sfc-gh-gbloom) [reply src/snowflake/cli/_plugins/workspace/action_context.py:18]: Importing `EntityBase` here introduces a dependency cycle, unfortunately

**Review by sfc-gh-fcampbell — APPROVED** [review]

**Pull request merged**